src/vision/shiny_future/barbara_makes_a_wish.md

```diff
@@ -0,0 +1,166 @@
+
+Barbara happily moves along to some initial performance analysis of her "working" code, eager to see what other things `wish4-async-insight` will reveal during her explorations.
+
+### Alternate History
+
+*The original status quo story just said that Barbara's problem was resolved (sort of) by switching to an unbounded channel. I, much like Barbara, could not tell *why* this resolved her problem. In particular, I could not tell whether there was an outright deadlock due to a cycle in the task-resource dependency chain that, or if there something more subtle happening. In the story above, I assumed it was the second case: something subtle.*
+
+*Here's an important alternate history though, for the first case of a cycle. Its ...* the same story, right up to when Barbara first runs `consolation`:
+
+```
+% rustup install wish4-consolation
+...
+% consolation --port 8080
+```
+
+This brings up a terminal window that looks similar to the Unix `top` program, except that instead of a list of OS processes, this offers a list of tasks, and shows their status (i.e. running, ready to poll, or blocked), as well as some metrics about how long the tasks spend in each state.
+
+At the top of the screen, Barbara sees highlighted warning: "deadlock cycle was detected. hit `P` for more info."
+
+Barbara types capital `P`. The terminal switches to "problem view," which shows
+
+ * The task types, ID, and attributes for each type.
+ * The resources being awaited on
+ * The location / backtrace of the await.
+ * A link to a documentation page expanding on the issue.
+
+The screen also says "hit `D` to generate a graphviz `.dot` file to disk describing the cycle."
+
+Barbara hits `D` and stares at the resulting graph.
+
+Barbara suddenly realizes her mistake: She had constructed a single task that was sometimes enqueuing work (by sending messages on the channel), and sometimes dequeuing work, but she had not put any controls into place to ensure that the dequeuing (via `recv`) would get prioritized as the channel filled up.
+
+Barbara reflects on the matter: she knows that she could swap in an unbounded channel to resolve this, but she thinks that she would be better off thinking a bit more about her system design, to see if she can figure out a way to supply back-pressure so that the send rate will go down as the channel fills up.
+
+
+## 🤔 Frequently Asked Questions
+
+### **What status quo story or stories are you retelling?**
+
+[Barbara wants Async Insights](../status_quo/barbara_wants_async_insights.md)
+
+### **What is [Alan] most excited about in this future? Is he disappointed by anything?**
+
+Alan is happy to see a tool that gives one a view into the internals of the async executor.
+
+Alan is not so thrilled about using the `consolation` terminal interface; but luckily there are other options, namely a web-browser based client that offers even richer functionality, such as renderings of the task/resource dependency graph.
```

Comment:
Can we also mention integrated editor extensions that provide this functionality as well?

Reply:
Fixed in 0eaf2add9434d720ce0732df07e6ded8188e3f2c